kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/ModemManagerDbusWrapper.java

```diff
@@ -250,119 +248,69 @@ private List<Properties> getBearersPropertiesFromPaths(List<DBusPath> bearerPath
 
     }
 
-    protected void resetHandlerEnable(String deviceId, Optional<String> modemManagerDbusPath, int delayMinutes,
-            String networkManagerDbusPath) throws DBusException {
-        if (!modemManagerDbusPath.isPresent()) {
-            logger.warn("Cannot retrieve modem device for {}. Skipping modem reset monitor setup.", deviceId);
+    protected void modemTaskHandler(String deviceId, Device device, NetworkProperties properties) throws DBusException {
```

Comment:
Three things here:
1. The name `modemTaskHandler` is not really significative in this context and, given we have a class with the same name, it's quite confusing. This method essentially sets up the signal handler responsible for monitoring the device... therefore this performs the opposite of `modemTaskHandlerDisable`. I would call this `modemTaskHandlerEnable`.
2. `ModemManagerDbusWrapper` contains all the code that directly interfaces with Modem Manager. These "TaskHandlers" are mainly working with Network Manager though: they use NM signals and NM data structure. Is this the correct place for this code?
3. Finally: it looks like this "Task Handler" generator could be promoted to its own object 🤔  Is there a reason why you put it inside the `ModemManagerDbusWrapper`?